syntax = "proto3";

package app;

service WordGame {
    rpc CreateGame(GameRequest) returns (GameResponse) {}
    rpc InitGameRequest(InitRequest) returns (InitResponse) {}
    rpc SubmitWord(WordSubmissionRequest) returns (WordSubmissionResponse) {}
}

message GameRequest {
    string gameType = 1;
}

message GameResponse {
    bytes gameId = 1;
}

message InitRequest {
    bytes gameId = 1;
}

message InitResponse {
    string letters = 1;
}

message WordSubmissionRequest {
    bytes gameId = 1;
    string word = 2;
}

message WordSubmissionResponse {
    int32 score = 1;
    int32 total = 2;
    string message = 3;
}